### DISCUSSION TITLE
stale queries

### DISCUSSION THREAD

**Discussion by orangepineapple** [ANLAB-KAIST/KENSv3#252] [Project Q&A 3. Internet Layer (IP/Routing)]
When a router initially starts off in the network with an empty it will take some time to build the shortest path to all other routers, and that shortest path will change overtime. As such, is the pwosfquery supposed to return us the shortest path (-1 if the path has not been propagated yet) at the point in time it is called. Or is there supposed to be some mechanism to weight for the router to get a somewhat "decent" shorted path.

**Comment by Pusnow** [comment]
> As such, is the pwosfquery supposed to return us the shortest path (-1 if the path has not been propagated yet) at the point in time it is called. Or is there supposed to be some mechanism to weight for the router to get a somewhat "decent" shorted path.

You should return shortest path at that point, but we will give enough time to converge.
